src/runtime/prte_finalize.c

```diff
@@ -74,7 +75,33 @@ int prte_finalize(void)
     /* release the cache */
     PRTE_RELEASE(prte_cache);
 
-    /* release the job hash table; pop the first element in key order and release it, repeat */
+    prte_display_prte_job_data("Finalize");
+
+    /* Release the job hash table
+     *
+     * First we need to iterate through an release all of the children.
+     * Since we cannot know the order in which items are removed from the hash
+     * table, then if we do not do this then we may try to free a child job
+     * before the parent which would trigger an assert in the list item
+     * destructor which checks to make sure this item was removed from the list
+     * before being destructed.
+     *
+     * Second (and not with the first loop) we can then destruct the
+     * jobs in the hash table knowing that they are definitely not on a
+     * children list.
+     */
+    PRTE_HASH_TABLE_FOREACH(key, uint32, jdata, prte_job_data) {
+        if (NULL != jdata) {
+            // Remove all children from the list
+            // We do not want to destruct this list here since that occurs in the
+            // prte_job_t destructor - which will happen in the next loop.
+            PRTE_LIST_FOREACH(child_jdata, &jdata->children, prte_job_t) {
```

Comment:
This needs to be PRTE_LIST_FOREACH_SAFE since you are modifying the list in the loop

Reply:
Thanks. I'll fix.

Reply:
Done.